.github/workflows/python-package.yml

```diff
@@ -28,4 +28,8 @@ jobs:
         flake8 . --count --max-complexity=10 --max-line-length=127 --statistics
     - name: Test with pytest
       run: |
-        echo Add your testing here
+        pytest -rp
```

Comment:
Nice.
A question: how do we save test results to a file and send to Slack?

Reply:
BTW, why -rp instead of -v?

Reply:
Echo the results to slack bot with webhooks , I might try [How to Send a Message to Slack From a Bash Script](https://www.cloudsavvyit.com/289/how-to-send-a-message-to-slack-from-a-bash-script/)

Reply:
Yes, that's one option.
Another one is using any one of the existing Github actions for Slack:
https://github.com/marketplace/actions/github-action-for-slack#alternatives

Reply:
Why -rp? It was in a manual I've found, not sure which one

Reply:
Gotcha!
In this case it works fine but running `pytest --help` we see:

> -r chars              show extra test summary info as specified by chars: (f)ailed, (E)rror, (s)kipped, (x)failed,
                        (X)passed, (p)assed, (P)assed with output, (a)ll except passed (p/P), or (A)ll. (w)arnings are
                        enabled by default (see --disable-warnings), 'N' can be used to reset the list. (default:
                        'fE').

Which means this only shows extra info for passed tests, not failed or skipped ones.
In CI we are usually especially interested in failed tests. So adding -rpx would be better. Or just going full cycle on `-v`